internal/feature/flags.go

```diff
@@ -0,0 +1,59 @@
+package feature
+
+import (
+	"fmt"
+	"sync"
+)
+
+// All feature flags need to be defined here with their default values
+var flags = map[string]bool{
+	"events": false,
+}
+
+type Checker interface {
+	IsEnabled(flag string) bool
+}
+
+type Writer interface {
+	Enable(flag string) error
+	Disable(flag string) error
+}
+
+func NewStaticMapFeature() *staticMapFeature {
+	return &staticMapFeature{flags: flags, mu: &sync.Mutex{}}
+}
+
+type staticMapFeature struct {
+	flags map[string]bool
+	mu    *sync.Mutex
+}
+
+func (f *staticMapFeature) IsEnabled(flag string) bool {
+	f.mu.Lock()
+	defer f.mu.Unlock()
+	return f.flags[flag]
```

Comment:
what happens if we typo a flag name in IsEnabled?

Reply:
It will return false, but I think it should return an error. Good call.